back/app/services/admin_publications_filtering_service.rb

```diff
@@ -51,4 +53,14 @@ class AdminPublicationsFilteringService
   add_filter('top_level_only') do |scope, options|
     [0, '0'].include?(options[:depth]) ? scope.where(depth: 0) : scope
   end
+
+  add_filter('remove_childless_parents') do |scope, options|
+    next scope unless ['true', true, '1'].include? options[:remove_childless_parents]
+
+    parents_of_project_publications_ids = project_publications.where.not(parent_id: nil).map(&:parent_id)
+    parents_of_project_publications     = scope.where(id: parents_of_project_publications_ids)
+    non_parents                         = scope.where(children_allowed: false)
+    
+    parents_of_project_publications.or(non_parents)
```

Comment:
What do you think about?
```suggestion
    parents_with_children = scope.where(id: scope.select(:parent_id).where.not(parent_id: nil).distinct)
    non_parents = scope.where(children_allowed: false)
    parents_with_children.or(non_parents)
```